### ISSUE TITLE
CodeGen does not return $lastexitcode 1 when error happens and erros are not written to StdErr

### ISSUE THREAD

**Issue by leandrodotec** [Avanade/Beef#71]
**Expected behavirour:**

When an error happens, a $lastexitcode should be returned and error message should be written to stdErr. At the moment it returns 0 when it succeeds or errors. 

**Extra background:**

This will allow running code gen in automated builds to check for things like:

- Changes to xml files have been committed without committing generated files. 
- Changes to database were made without re-generating stored procedures.

**Occurs in version:**  3.1.20

**How to reproduce:**
In Powershell:
```
PS>dotnet run all --expectNoChanges 
...
  Template: EntityGrpcServiceAgent_cs.xml (Common/Grpc/ServiceAgents)
   [Files: Unchanged = 0, Updated = 0, Created = 0]
An unexpected exception occurred during the 'Run' phase: Unexpected changes detected; one or more files were created and/or updated. Exception: Beef.CodeGen.CodeGenException: Unexpected changes detected; one or more files were created and/or updated.
   at Beef.CodeGen.CodeGenExecutor.OnRunAsync(ExecutorRunArgs args)
   at Beef.Executors.ExecutorBase.RunAsync(ExecutorRunArgs args)
Executor '0e879bbc-8ef7-4841-ac24-f20d0329fe0e' stop as a result of an unhandled exception: Unexpected changes detected; one or more files were created and/or updated. Exception: Beef.CodeGen.CodeGenException: Unexpected changes detected; one or more files were created and/or updated.
   at Beef.CodeGen.CodeGenExecutor.OnRunAsync(ExecutorRunArgs args)
   at Beef.Executors.ExecutorBase.RunAsync(ExecutorRunArgs args)
   at Beef.Executors.ExecutionManager.<>c__DisplayClass17_1.<<Create>b__2>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Beef.Executors.Executor.<>c__DisplayClass34_0.<<RunWrapperAsync>b__0>d.MoveNext()

PS> $lastexitcode
0
```

or

bring database down and run:
dotnet run database

**Comment by chullybun** [comment]
Hi @leandrodotec - sorry for the delay. As an FYI, I have fixed this issue in the latest branch that I am working on which I will merge and publish in the next couple of days. We will need to work with @edjo23 on the timing of applying as this next update will contain all of the entity code-gen changes as part of the migration to handlebars (the new code-gen templating engine). Cheers...

**Comment by chullybun** [comment]
The latest version has just been published to nuget; get `v4.1.2` (or later) of `Beef.CodeGen.Core`.

**Comment by leandrodotec** [comment]
Thank you. 
We are in the process of upgrading our codebase to v4+.
Cheers,
